Readme.md

```diff
@@ -1,5 +1,262 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+Before we start to define anything for the lexer you will need to learn the syntax for [BNF](#bnf)
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set up some token culling rules
+
+Here is an example of what our BNF rules look like
+```
+root ::= NumberList
+sep ::= " "
+number ::= [0-9]+
+NumberList ::= number (sep number)*
+```
+
+Token input
+`1 2 3`
+
+The token culling rules tell the lexer what do with tokens with the given names, allowing the lexer to remove tokens that match the token name before returning the token tree with them in it. 
```

Comment:
Probably better to show the output before you explain what culling rules are and show how to define them.